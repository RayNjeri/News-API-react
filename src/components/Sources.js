import React, { Component } from 'react';
import ReactDOM from 'react-dom';
import { BrowserRouter as Router, Route, Link } from 'react-router-dom';

import Axios from 'axios';

export default class Sources extends Component {
    constructor(props) {
        super(props);
        this.state = {
            sources: [],
        };
    }

    componentDidMount() {
        Axios.get('https://newsapi.org/v1/sources')
            .then((result) => {
                console.log('getting result: ', result);
                this.setState({
                    sources: result.data.sources
                });
            });
    }
    render() {
        return (
            < div >
                <h2>SOURCES</h2>
                {this.state.sources.map((sources) => (
                    <div>
                        <a href={sources.url}>{sources.name}</a>
                    </div>
                ))
                }
            </div >
        );
    }
};

