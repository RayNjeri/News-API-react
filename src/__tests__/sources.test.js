import React from 'react';
import React from 'react-addons';
import ReactTestUtils from 'react-dom/test-utils';
import { mount, shallow, wrapper } from 'enzyme';
import shallowToJSON from 'enzyme-to-json';
import Sources from '../components/Sources.js';

describe('Sources components', () => {
    beforeEach, () => {
        this.exampleSources = [
            { id: 1, name: "ABC News(AU)" },
            { id: 2, name: "Al Jazeera English" }
        ];
    };
    it('snapshot for sources component', () => {
        const component = shallow(<Sources data={[]} />);
        const tree = shallowToJSON(component);
        expect(tree).toMatchSnapshot();
    });
    it('renders a list of available news sources', () => {
        let y = TestUtils.renderIntoDocument(<this.state.sources.map />);
        expect(y.refs.getDOMNode().textContent).toEqual(''); expect(y.refs.list).toBeUndefined();
    });

}
