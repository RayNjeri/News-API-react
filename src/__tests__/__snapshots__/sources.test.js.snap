// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Sources Component snapshot for sources component 1`] = `
<div>
  <h1>
    SOURCES
  </h1>
</div>
`;
